crates/starknet-devnet/tests/test_fork.rs

```diff
@@ -112,12 +111,20 @@ mod fork_tests {
 
         let fork_devnet = origin_devnet.fork().await.unwrap();
 
-        let resp =
+        let resp_block_hash =
             &fork_devnet.json_rpc_client.get_block_with_tx_hashes(BlockId::Hash(block_hash)).await;
+        match resp_block_hash {
+            Ok(MaybePendingBlockWithTxHashes::Block(b)) => assert_eq!(b.block_number, 1),
+            _ => panic!("Unexpected resp: {resp_block_hash:?}"),
+        };
 
-        match resp {
-            Ok(MaybePendingBlockWithTxHashes::Block(b)) => assert_eq!(b.block_number, 0),
-            _ => panic!("Unexpected resp: {resp:?}"),
+        let resp_latest_block = &fork_devnet
+            .json_rpc_client
+            .get_block_with_tx_hashes(BlockId::Tag(BlockTag::Latest))
+            .await;
+        match resp_latest_block {
+            Ok(MaybePendingBlockWithTxHashes::Block(b)) => assert_eq!(b.block_number, 2),
```

Comment:
Why is it 2? Origin has genesis (number 0) + one created block (1). So number 2 belongs to the genesis block of the fork? What about genesis block hash values?

Reply:
There is a genesis block in origin_devnet and later there is create_block(), later we fork it so it's new genesis block in fork (it's needed there to have block-state relation)

Reply:
To be clear, my first question from the previous comment is just making a check. Perhaps you can add a comment explaining that.

Reply:
I edited tests here e244880e6cb41c41b4d8a9312001cb6f2c245d94